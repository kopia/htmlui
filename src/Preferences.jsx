import { Component } from 'react';
import { ThemeSelector } from './ThemeSelector';
import { UIPreferencesContext } from './contexts/UIPreferencesContext';

export class Preferences extends Component {
    static contextType = UIPreferencesContext;
    constructor() {
        super();
        this.state = {
            status: {},
            error: null,
        };
    }

    render() {
        const { pageSize, bytesStringBase2 } = this.context;
        return <>
            <form>
                <div className='form-group'>
                    <label className='label-description' id='themeLabel'>Theme</label>
                    <ThemeSelector />
                    <small htmlFor='themeLabel' id='themeHelp' className='form-text text-muted'>The current active theme</small>
                </div>
                <br />
                <div className='form-group'>
                    <label className='label-description'>Bytes to the base of 2</label>
                    <input className='form-control form-control-sm' id='bytesBaseInput'
                        type='text' value={bytesStringBase2} disabled={true} />
                    <small htmlFor='bytesBaseInput' id='bytesHelp' className='form-text text-muted'>Represents bytes to the base of 2</small>
                </div>
                <br />
                <div className='form-group'>
                    <label className='label-description'>Page size</label>
                    <input className='form-control form-control-sm' id='pageSizeInput'
                        type='text' placeholder='Page size' value={pageSize} disabled={true} />
                    <small hmtlFor="pageSizeInput" id='pageSizeHelp' className='form-text text-muted'>The number of items shown in tables</small>
                </div>
            </form>
        </>
    }
}