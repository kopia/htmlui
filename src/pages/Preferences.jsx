import { useContext } from 'react';
import { useTranslation } from "react-i18next";
import { UIPreferencesContext } from '../contexts/UIPreferencesContext';

/**
 * Class that exports preferences
 */
export function Preferences() {
    const { t } = useTranslation();
    const { pageSize, theme, bytesStringBase2, fontSize, setFontSize, setByteStringBase, setTheme } = useContext(UIPreferencesContext);
    return <>
        <form>
            <div className='form-group'>
                <label className='label-description' htmlFor='themeSelector' id='themeLabel'>{t('feedback.ui.theme-description')}</label>
                <select className="form-select form-select-sm" title={t('feedback.ui.theme-select')} id='themeSelector' value={theme} onChange={e => setTheme(e.target.value)}>
                    <option value="light">{t('value.ui.theme-light')}</option>
                    <option value="dark">{t('value.ui.theme-dark')}</option>
                    <option value="pastel">{t('value.ui.theme-pastel')}</option>
                    <option value="ocean">{t('value.ui.theme-ocean')}</option>
                </select>
                <small hmtlfor='themeLabel' id='themeHelp' className='form-text text-muted'>{t('feedback.ui.theme-help')}</small>
            </div>
            <br />
            <div className='form-group'>
                <label className='label-description' htmlFor="bytesBaseInput">{t('feedback.ui.byte-representation-description')}</label>
                <select className="form-select form-select-sm" title={t('feedback.ui.byte-representation-select')} id='bytesBaseInput' value={bytesStringBase2} onChange={e => setByteStringBase(e.target.value)}>
                    <option value="true">{t('value.ui.byte-representation-base2')}</option>
                    <option value="false">{t('value.ui.byte-representation-base10')}</option>
                </select>
                <small hmtlfor='bytesBaseInput' id='bytesHelp' className='form-text text-muted'>{t('feedback.ui.byte-representation-help')}</small>
            </div>
            <br />
            <div className='form-group'>
                <label className='label-description'>{t('feedback.ui.appearance')}</label>
                <select className="form-select form-select-sm" title={t('feedback.ui.appearance-hint')} id='appearanceInput' value={fontSize} onChange={e => setFontSize(e.target.value)}>
                    <option value="fs-6">{t('value.ui.appearance-small')}</option>
                    <option value="fs-5">{t('value.ui.appearance-medium')}</option>
                    <option value="fs-4">{t('value.ui.appearance-large')}</option>
                </select>
                <small hmtlfor="appearanceInput" id='appearanceHelp' className='form-text text-muted'>{t('feedback.ui.appearance-help')}</small>
            </div>
            <br />
            <div className='form-group'>
                <label className='label-description'>{t('feedback.ui.pagesize-description')}</label>
                <input className='form-control form-control-sm' id='pageSizeInput'
                    type='text' placeholder={t('feedback.ui.pagesize-hint')} value={pageSize} disabled={true} />
                <small hmtlfor="pageSizeInput" id='pageSizeHelp' className='form-text text-muted'>{t('feedback.ui.pagesize-help')}</small>
            </div>
        </form >
    </>
}