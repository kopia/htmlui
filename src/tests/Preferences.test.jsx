import { render, screen } from '@testing-library/react'
import { expect, test } from '@jest/globals';
import userEvent from "@testing-library/user-event";
import { Preferences } from '../pages/Preferences';

import { configure } from '@testing-library/dom'
configure({ testIdAttribute: 'id' })

// Wrapper
let wrapper;

/**
 * 
 */
beforeEach(() => {
    wrapper = render(<Preferences />)
});

/**
 * 
 */
describe('Calling the preference page', () => {
    test('Should render preferences', () => {
        expect(wrapper).toMatchSnapshot();
    })
})

/**
 * 
 */
describe('Select the light theme', () => {
    test('Should select light theme', () => {
     userEvent.selectOptions(
            screen.getByTestId('themeSelector'),
            screen.getByRole('option', { name: "value.ui.theme-light" }));

        expect(screen.getByRole('option', { name: "value.ui.theme-light" }).selected).toBe(true)
        expect(wrapper).toMatchSnapshot();
    })
})

/**
 * 
 */
describe('Test number of themes', () => {
    test('Should have four themes', () => {
        let selector = screen.getByTestId('themeSelector');
        expect(selector).toHaveLength(4);
        expect(wrapper).toMatchSnapshot();
    })
})

/**
 * 
 */
describe('Test byte representation', () => {
    test('Should have two options', () => {
        let selector = screen.getByTestId('bytesBaseInput');
        expect(selector).toHaveLength(2);
        expect(wrapper).toMatchSnapshot();
    })
})