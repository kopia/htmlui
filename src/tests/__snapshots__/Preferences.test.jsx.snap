// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calling the preference page Should render preferences 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="themeSelector"
            id="themeLabel"
          >
            feedback.ui.theme-description
          </label>
          <select
            class="form-select form-select-sm"
            id="themeSelector"
            title="feedback.ui.theme-select"
          >
            <option
              value="light"
            >
              value.ui.theme-light
            </option>
            <option
              value="dark"
            >
              value.ui.theme-dark
            </option>
            <option
              value="pastel"
            >
              value.ui.theme-pastel
            </option>
            <option
              value="ocean"
            >
              value.ui.theme-ocean
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="themeLabel"
            id="themeHelp"
          >
            feedback.ui.theme-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="bytesBaseInput"
          >
            feedback.ui.byte-representation-description
          </label>
          <select
            class="form-select form-select-sm"
            id="bytesBaseInput"
            title="feedback.ui.byte-representation-select"
          >
            <option
              value="true"
            >
              value.ui.byte-representation-base2
            </option>
            <option
              value="false"
            >
              value.ui.byte-representation-base10
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="bytesBaseInput"
            id="bytesHelp"
          >
            feedback.ui.byte-representation-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.appearance
          </label>
          <select
            class="form-select form-select-sm"
            id="appearanceInput"
            title="feedback.ui.appearance-hint"
          >
            <option
              value="fs-6"
            >
              value.ui.appearance-small
            </option>
            <option
              value="fs-5"
            >
              value.ui.appearance-medium
            </option>
            <option
              value="fs-4"
            >
              value.ui.appearance-large
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="appearanceInput"
            id="appearanceHelp"
          >
            feedback.ui.appearance-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.pagesize-description
          </label>
          <input
            class="form-control form-control-sm"
            disabled=""
            id="pageSizeInput"
            placeholder="feedback.ui.pagesize-hint"
            type="text"
            value=""
          />
          <small
            class="form-text text-muted"
            hmtlfor="pageSizeInput"
            id="pageSizeHelp"
          >
            feedback.ui.pagesize-help
          </small>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="themeSelector"
          id="themeLabel"
        >
          feedback.ui.theme-description
        </label>
        <select
          class="form-select form-select-sm"
          id="themeSelector"
          title="feedback.ui.theme-select"
        >
          <option
            value="light"
          >
            value.ui.theme-light
          </option>
          <option
            value="dark"
          >
            value.ui.theme-dark
          </option>
          <option
            value="pastel"
          >
            value.ui.theme-pastel
          </option>
          <option
            value="ocean"
          >
            value.ui.theme-ocean
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="themeLabel"
          id="themeHelp"
        >
          feedback.ui.theme-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="bytesBaseInput"
        >
          feedback.ui.byte-representation-description
        </label>
        <select
          class="form-select form-select-sm"
          id="bytesBaseInput"
          title="feedback.ui.byte-representation-select"
        >
          <option
            value="true"
          >
            value.ui.byte-representation-base2
          </option>
          <option
            value="false"
          >
            value.ui.byte-representation-base10
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="bytesBaseInput"
          id="bytesHelp"
        >
          feedback.ui.byte-representation-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.appearance
        </label>
        <select
          class="form-select form-select-sm"
          id="appearanceInput"
          title="feedback.ui.appearance-hint"
        >
          <option
            value="fs-6"
          >
            value.ui.appearance-small
          </option>
          <option
            value="fs-5"
          >
            value.ui.appearance-medium
          </option>
          <option
            value="fs-4"
          >
            value.ui.appearance-large
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="appearanceInput"
          id="appearanceHelp"
        >
          feedback.ui.appearance-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.pagesize-description
        </label>
        <input
          class="form-control form-control-sm"
          disabled=""
          id="pageSizeInput"
          placeholder="feedback.ui.pagesize-hint"
          type="text"
          value=""
        />
        <small
          class="form-text text-muted"
          hmtlfor="pageSizeInput"
          id="pageSizeHelp"
        >
          feedback.ui.pagesize-help
        </small>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select the light theme Should select light theme 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="themeSelector"
            id="themeLabel"
          >
            feedback.ui.theme-description
          </label>
          <select
            class="form-select form-select-sm"
            id="themeSelector"
            title="feedback.ui.theme-select"
          >
            <option
              value="light"
            >
              value.ui.theme-light
            </option>
            <option
              value="dark"
            >
              value.ui.theme-dark
            </option>
            <option
              value="pastel"
            >
              value.ui.theme-pastel
            </option>
            <option
              value="ocean"
            >
              value.ui.theme-ocean
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="themeLabel"
            id="themeHelp"
          >
            feedback.ui.theme-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="bytesBaseInput"
          >
            feedback.ui.byte-representation-description
          </label>
          <select
            class="form-select form-select-sm"
            id="bytesBaseInput"
            title="feedback.ui.byte-representation-select"
          >
            <option
              value="true"
            >
              value.ui.byte-representation-base2
            </option>
            <option
              value="false"
            >
              value.ui.byte-representation-base10
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="bytesBaseInput"
            id="bytesHelp"
          >
            feedback.ui.byte-representation-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.appearance
          </label>
          <select
            class="form-select form-select-sm"
            id="appearanceInput"
            title="feedback.ui.appearance-hint"
          >
            <option
              value="fs-6"
            >
              value.ui.appearance-small
            </option>
            <option
              value="fs-5"
            >
              value.ui.appearance-medium
            </option>
            <option
              value="fs-4"
            >
              value.ui.appearance-large
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="appearanceInput"
            id="appearanceHelp"
          >
            feedback.ui.appearance-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.pagesize-description
          </label>
          <input
            class="form-control form-control-sm"
            disabled=""
            id="pageSizeInput"
            placeholder="feedback.ui.pagesize-hint"
            type="text"
            value=""
          />
          <small
            class="form-text text-muted"
            hmtlfor="pageSizeInput"
            id="pageSizeHelp"
          >
            feedback.ui.pagesize-help
          </small>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="themeSelector"
          id="themeLabel"
        >
          feedback.ui.theme-description
        </label>
        <select
          class="form-select form-select-sm"
          id="themeSelector"
          title="feedback.ui.theme-select"
        >
          <option
            value="light"
          >
            value.ui.theme-light
          </option>
          <option
            value="dark"
          >
            value.ui.theme-dark
          </option>
          <option
            value="pastel"
          >
            value.ui.theme-pastel
          </option>
          <option
            value="ocean"
          >
            value.ui.theme-ocean
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="themeLabel"
          id="themeHelp"
        >
          feedback.ui.theme-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="bytesBaseInput"
        >
          feedback.ui.byte-representation-description
        </label>
        <select
          class="form-select form-select-sm"
          id="bytesBaseInput"
          title="feedback.ui.byte-representation-select"
        >
          <option
            value="true"
          >
            value.ui.byte-representation-base2
          </option>
          <option
            value="false"
          >
            value.ui.byte-representation-base10
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="bytesBaseInput"
          id="bytesHelp"
        >
          feedback.ui.byte-representation-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.appearance
        </label>
        <select
          class="form-select form-select-sm"
          id="appearanceInput"
          title="feedback.ui.appearance-hint"
        >
          <option
            value="fs-6"
          >
            value.ui.appearance-small
          </option>
          <option
            value="fs-5"
          >
            value.ui.appearance-medium
          </option>
          <option
            value="fs-4"
          >
            value.ui.appearance-large
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="appearanceInput"
          id="appearanceHelp"
        >
          feedback.ui.appearance-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.pagesize-description
        </label>
        <input
          class="form-control form-control-sm"
          disabled=""
          id="pageSizeInput"
          placeholder="feedback.ui.pagesize-hint"
          type="text"
          value=""
        />
        <small
          class="form-text text-muted"
          hmtlfor="pageSizeInput"
          id="pageSizeHelp"
        >
          feedback.ui.pagesize-help
        </small>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test byte representation Should have two options 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="themeSelector"
            id="themeLabel"
          >
            feedback.ui.theme-description
          </label>
          <select
            class="form-select form-select-sm"
            id="themeSelector"
            title="feedback.ui.theme-select"
          >
            <option
              value="light"
            >
              value.ui.theme-light
            </option>
            <option
              value="dark"
            >
              value.ui.theme-dark
            </option>
            <option
              value="pastel"
            >
              value.ui.theme-pastel
            </option>
            <option
              value="ocean"
            >
              value.ui.theme-ocean
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="themeLabel"
            id="themeHelp"
          >
            feedback.ui.theme-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="bytesBaseInput"
          >
            feedback.ui.byte-representation-description
          </label>
          <select
            class="form-select form-select-sm"
            id="bytesBaseInput"
            title="feedback.ui.byte-representation-select"
          >
            <option
              value="true"
            >
              value.ui.byte-representation-base2
            </option>
            <option
              value="false"
            >
              value.ui.byte-representation-base10
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="bytesBaseInput"
            id="bytesHelp"
          >
            feedback.ui.byte-representation-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.appearance
          </label>
          <select
            class="form-select form-select-sm"
            id="appearanceInput"
            title="feedback.ui.appearance-hint"
          >
            <option
              value="fs-6"
            >
              value.ui.appearance-small
            </option>
            <option
              value="fs-5"
            >
              value.ui.appearance-medium
            </option>
            <option
              value="fs-4"
            >
              value.ui.appearance-large
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="appearanceInput"
            id="appearanceHelp"
          >
            feedback.ui.appearance-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.pagesize-description
          </label>
          <input
            class="form-control form-control-sm"
            disabled=""
            id="pageSizeInput"
            placeholder="feedback.ui.pagesize-hint"
            type="text"
            value=""
          />
          <small
            class="form-text text-muted"
            hmtlfor="pageSizeInput"
            id="pageSizeHelp"
          >
            feedback.ui.pagesize-help
          </small>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="themeSelector"
          id="themeLabel"
        >
          feedback.ui.theme-description
        </label>
        <select
          class="form-select form-select-sm"
          id="themeSelector"
          title="feedback.ui.theme-select"
        >
          <option
            value="light"
          >
            value.ui.theme-light
          </option>
          <option
            value="dark"
          >
            value.ui.theme-dark
          </option>
          <option
            value="pastel"
          >
            value.ui.theme-pastel
          </option>
          <option
            value="ocean"
          >
            value.ui.theme-ocean
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="themeLabel"
          id="themeHelp"
        >
          feedback.ui.theme-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="bytesBaseInput"
        >
          feedback.ui.byte-representation-description
        </label>
        <select
          class="form-select form-select-sm"
          id="bytesBaseInput"
          title="feedback.ui.byte-representation-select"
        >
          <option
            value="true"
          >
            value.ui.byte-representation-base2
          </option>
          <option
            value="false"
          >
            value.ui.byte-representation-base10
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="bytesBaseInput"
          id="bytesHelp"
        >
          feedback.ui.byte-representation-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.appearance
        </label>
        <select
          class="form-select form-select-sm"
          id="appearanceInput"
          title="feedback.ui.appearance-hint"
        >
          <option
            value="fs-6"
          >
            value.ui.appearance-small
          </option>
          <option
            value="fs-5"
          >
            value.ui.appearance-medium
          </option>
          <option
            value="fs-4"
          >
            value.ui.appearance-large
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="appearanceInput"
          id="appearanceHelp"
        >
          feedback.ui.appearance-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.pagesize-description
        </label>
        <input
          class="form-control form-control-sm"
          disabled=""
          id="pageSizeInput"
          placeholder="feedback.ui.pagesize-hint"
          type="text"
          value=""
        />
        <small
          class="form-text text-muted"
          hmtlfor="pageSizeInput"
          id="pageSizeHelp"
        >
          feedback.ui.pagesize-help
        </small>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test number of themes Should have four themes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="themeSelector"
            id="themeLabel"
          >
            feedback.ui.theme-description
          </label>
          <select
            class="form-select form-select-sm"
            id="themeSelector"
            title="feedback.ui.theme-select"
          >
            <option
              value="light"
            >
              value.ui.theme-light
            </option>
            <option
              value="dark"
            >
              value.ui.theme-dark
            </option>
            <option
              value="pastel"
            >
              value.ui.theme-pastel
            </option>
            <option
              value="ocean"
            >
              value.ui.theme-ocean
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="themeLabel"
            id="themeHelp"
          >
            feedback.ui.theme-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
            for="bytesBaseInput"
          >
            feedback.ui.byte-representation-description
          </label>
          <select
            class="form-select form-select-sm"
            id="bytesBaseInput"
            title="feedback.ui.byte-representation-select"
          >
            <option
              value="true"
            >
              value.ui.byte-representation-base2
            </option>
            <option
              value="false"
            >
              value.ui.byte-representation-base10
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="bytesBaseInput"
            id="bytesHelp"
          >
            feedback.ui.byte-representation-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.appearance
          </label>
          <select
            class="form-select form-select-sm"
            id="appearanceInput"
            title="feedback.ui.appearance-hint"
          >
            <option
              value="fs-6"
            >
              value.ui.appearance-small
            </option>
            <option
              value="fs-5"
            >
              value.ui.appearance-medium
            </option>
            <option
              value="fs-4"
            >
              value.ui.appearance-large
            </option>
          </select>
          <small
            class="form-text text-muted"
            hmtlfor="appearanceInput"
            id="appearanceHelp"
          >
            feedback.ui.appearance-help
          </small>
        </div>
        <br />
        <div
          class="form-group"
        >
          <label
            class="label-description"
          >
            feedback.ui.pagesize-description
          </label>
          <input
            class="form-control form-control-sm"
            disabled=""
            id="pageSizeInput"
            placeholder="feedback.ui.pagesize-hint"
            type="text"
            value=""
          />
          <small
            class="form-text text-muted"
            hmtlfor="pageSizeInput"
            id="pageSizeHelp"
          >
            feedback.ui.pagesize-help
          </small>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="themeSelector"
          id="themeLabel"
        >
          feedback.ui.theme-description
        </label>
        <select
          class="form-select form-select-sm"
          id="themeSelector"
          title="feedback.ui.theme-select"
        >
          <option
            value="light"
          >
            value.ui.theme-light
          </option>
          <option
            value="dark"
          >
            value.ui.theme-dark
          </option>
          <option
            value="pastel"
          >
            value.ui.theme-pastel
          </option>
          <option
            value="ocean"
          >
            value.ui.theme-ocean
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="themeLabel"
          id="themeHelp"
        >
          feedback.ui.theme-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
          for="bytesBaseInput"
        >
          feedback.ui.byte-representation-description
        </label>
        <select
          class="form-select form-select-sm"
          id="bytesBaseInput"
          title="feedback.ui.byte-representation-select"
        >
          <option
            value="true"
          >
            value.ui.byte-representation-base2
          </option>
          <option
            value="false"
          >
            value.ui.byte-representation-base10
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="bytesBaseInput"
          id="bytesHelp"
        >
          feedback.ui.byte-representation-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.appearance
        </label>
        <select
          class="form-select form-select-sm"
          id="appearanceInput"
          title="feedback.ui.appearance-hint"
        >
          <option
            value="fs-6"
          >
            value.ui.appearance-small
          </option>
          <option
            value="fs-5"
          >
            value.ui.appearance-medium
          </option>
          <option
            value="fs-4"
          >
            value.ui.appearance-large
          </option>
        </select>
        <small
          class="form-text text-muted"
          hmtlfor="appearanceInput"
          id="appearanceHelp"
        >
          feedback.ui.appearance-help
        </small>
      </div>
      <br />
      <div
        class="form-group"
      >
        <label
          class="label-description"
        >
          feedback.ui.pagesize-description
        </label>
        <input
          class="form-control form-control-sm"
          disabled=""
          id="pageSizeInput"
          placeholder="feedback.ui.pagesize-hint"
          type="text"
          value=""
        />
        <small
          class="form-text text-muted"
          hmtlfor="pageSizeInput"
          id="pageSizeHelp"
        >
          feedback.ui.pagesize-help
        </small>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
